From e1e8734b5698b7d4c9f7500feee9e3e92f8f4f2f Mon Sep 17 00:00:00 2001
From: Tianshu Qiu <tian.shu.qiu@intel.com>
Date: Fri, 5 Dec 2014 16:23:17 +0800
Subject: [PATCH 266/269] lm3554: Remove sub device from platform data when
 unloading

This patch removes the v4l2 sub device of LM3554
from platform data when driver is unloading.

Change-Id: I298c1e15baec494ded84c2f56adf4fcd9d00d309
Tracked-On:https://jira01.devtools.intel.com/browse/IMINAN-12256
Signed-off-by: Tianshu Qiu <tian.shu.qiu@intel.com>
Reviewed-on: https://icggerrit.ir.intel.com/31699
Integrator: Wei Tao <wei.tao@intel.com>
Reviewed-by: Jing Tao <jing.tao@intel.com>
Approver: Wei Tao <wei.tao@intel.com>
Build: Wei Tao <wei.tao@intel.com>
Reviewed-by: Wei Tao <wei.tao@intel.com>
Maintainer: Wei Tao <wei.tao@intel.com>
---
 drivers/external_drivers/camera/drivers/media/i2c/lm3554.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/drivers/external_drivers/camera/drivers/media/i2c/lm3554.c b/drivers/external_drivers/camera/drivers/media/i2c/lm3554.c
index 165766d..1f1d919 100644
--- a/drivers/external_drivers/camera/drivers/media/i2c/lm3554.c
+++ b/drivers/external_drivers/camera/drivers/media/i2c/lm3554.c
@@ -933,6 +933,9 @@ static int lm3554_remove(struct i2c_client *client)
 
 	media_entity_cleanup(&flash->sd.entity);
 	v4l2_device_unregister_subdev(sd);
+#ifdef CONFIG_GMIN_INTEL_MID
+	atomisp_gmin_remove_subdev(sd);
+#endif
 
 	del_timer_sync(&flash->flash_off_delay);
 
-- 
1.9.1

